face/src/data/testcase/TestcaseModel.js

```diff
@@ -11,16 +12,15 @@ export default class TestcaseModel extends Model {
     }
 
     updateAttachments() {
-        function orEmpty(items) {
-            return items ? items : [];
+        function collectAttachments({steps, attachments}) {
+            return makeArray(steps)
+                .reduce((result, step) => result.concat(collectAttachments(step)), [])
+                .concat(makeArray(attachments));
         }
-        function collectAttachments(steps) {
-            return orEmpty(steps).reduce((attachments, step) => {
-                return attachments.concat(collectAttachments(step.steps), orEmpty(step.attachments));
-            }, []);
-        }
-        this.allAttachments = collectAttachments(this.get('steps'))
-            .concat(orEmpty(this.get('attachments')));
+        this.allAttachments = makeArray(this.get('beforeStages'))
```

Comment:
You are always wrapping this value into `makeArray`. Would be better to do it only once in `model.parse` and then always trust that value.